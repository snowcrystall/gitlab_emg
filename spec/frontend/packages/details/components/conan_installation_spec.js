import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import ConanInstallation from '~/packages/details/components/conan_installation.vue';
import InstallationTitle from '~/packages/details/components/installation_title.vue';
import CodeInstructions from '~/vue_shared/components/registry/code_instruction.vue';
import { conanPackage as packageEntity } from '../../mock_data';
import { registryUrl as conanPath } from '../mock_data';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('ConanInstallation', () => {
  let wrapper;

  const conanInstallationCommandStr = 'foo/command';
  const conanSetupCommandStr = 'foo/setup';

  const store = new Vuex.Store({
    state: {
      packageEntity,
      conanPath,
    },
    getters: {
      conanInstallationCommand: () => conanInstallationCommandStr,
      conanSetupCommand: () => conanSetupCommandStr,
    },
  });

  const findCodeInstructions = () => wrapper.findAll(CodeInstructions);
  const findInstallationTitle = () => wrapper.findComponent(InstallationTitle);

  function createComponent() {
    wrapper = shallowMount(ConanInstallation, {
      localVue,
      store,
    });
  }

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders all the messages', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('install command switch', () => {
    it('has the installation title component', () => {
      expect(findInstallationTitle().exists()).toBe(true);
      expect(findInstallationTitle().props()).toMatchObject({
        packageType: 'conan',
        options: [{ value: 'conan', label: 'Show Conan commands' }],
      });
    });
  });

  describe('installation commands', () => {
    it('renders the correct command', () => {
      expect(findCodeInstructions().at(0).props('instruction')).toBe(conanInstallationCommandStr);
    });
  });

  describe('setup commands', () => {
    it('renders the correct command', () => {
      expect(findCodeInstructions().at(1).props('instruction')).toBe(conanSetupCommandStr);
    });
  });
});
