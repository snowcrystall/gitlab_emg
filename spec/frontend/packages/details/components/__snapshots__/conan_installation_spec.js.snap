// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConanInstallation renders all the messages 1`] = `
<div>
  <installation-title-stub
    options="[object Object]"
    packagetype="conan"
  />
   
  <code-instruction-stub
    copytext="Copy Conan Command"
    instruction="foo/command"
    label="Conan Command"
    trackingaction="copy_conan_command"
    trackinglabel="code_instruction"
  />
   
  <h3
    class="gl-font-lg"
  >
    Registry setup
  </h3>
   
  <code-instruction-stub
    copytext="Copy Conan Setup Command"
    instruction="foo/setup"
    label="Add Conan Remote"
    trackingaction="copy_conan_setup_command"
    trackinglabel="code_instruction"
  />
   
  <gl-sprintf-stub
    message="For more information on the Conan registry, %{linkStart}see the documentation%{linkEnd}."
  />
</div>
`;
